iocore/net/UnixNetAccept.cc

```diff
@@ -201,37 +201,54 @@ NetAccept::init_accept(EThread *t)
   t->schedule_every(this, period);
 }
 
-void
-NetAccept::init_accept_per_thread()
+int
+NetAccept::accept_per_thread(int event, void *ep)
 {
-  int i, n;
-
-  ink_assert(opt.etype >= 0);
+  int listen_per_thread = 0;
+  REC_ReadConfigInteger(listen_per_thread, "proxy.config.exec_thread.listen");
 
-  if (do_listen(NON_BLOCKING)) {
-    return;
+  if (listen_per_thread == 1) {
+    if (do_listen(NON_BLOCKING)) {
+      return -1;
+    }
   }
 
   if (accept_fn == net_accept) {
     SET_HANDLER((NetAcceptHandler)&NetAccept::acceptFastEvent);
   } else {
     SET_HANDLER((NetAcceptHandler)&NetAccept::acceptEvent);
   }
+  PollDescriptor *pd = get_PollDescriptor(this_ethread());
+  if (this->ep.start(pd, this, EVENTIO_READ) < 0) {
+    Fatal("[NetAccept::init_accept_per_thread]:error starting EventIO");
+    return -1;
+  }
+  return 0;
+}
 
-  period = -HRTIME_MSECONDS(net_accept_period);
-  n      = eventProcessor.thread_group[opt.etype]._count;
+void
+NetAccept::init_accept_per_thread()
+{
+  int i, n;
+  int listen_per_thread = 0;
 
-  for (i = 0; i < n; i++) {
-    NetAccept *a       = (i < n - 1) ? clone() : this;
-    EThread *t         = eventProcessor.thread_group[opt.etype]._thread[i];
-    PollDescriptor *pd = get_PollDescriptor(t);
+  ink_assert(opt.etype >= 0);
+  REC_ReadConfigInteger(listen_per_thread, "proxy.config.exec_thread.listen");
 
-    if (a->ep.start(pd, a, EVENTIO_READ) < 0) {
-      Warning("[NetAccept::init_accept_per_thread]:error starting EventIO");
+  if (listen_per_thread == 0) {
+    if (do_listen(NON_BLOCKING)) {
+      return;
     }
+  }
+
+  SET_HANDLER((NetAcceptHandler)&NetAccept::accept_per_thread);
+  n = eventProcessor.thread_group[opt.etype]._count;
 
-    a->mutex = get_NetHandler(t)->mutex;
-    t->schedule_every(a, period);
```

Comment:
No, you can not remove this line.

Because:

- Our epoll is running in ET mode, which means that ATS MUST do accept() until EAGAIN occurs.
- But NetAccept::acceptFastEvent() does not meet the requirements.

Reply:
isn't acceptFastEvent doing that? If I am missing something, I would prefer to fix acceptFastEvent instead of scheduling accept periodically.

Reply:
https://github.com/apache/trafficserver/blob/15b0051fa2230bb60922943f4207b4a811f65430/iocore/net/UnixNetAccept.cc#L404-L407

@vmamidi If the ATS reaches the connection limit, acceptFastEvent will return immediately.

Reply:
That's a good find. I should have fixed that as part of #5305. I will handle the throttling scenario. Do you see any problems other than that?